Automattic/jetpack
# Sharing: Add X button

Fixes #32178
Fixes #33315

## Proposed changes:

This PR does 2 things:

1. It adds a new button for the X service.
2. It changes the default sharing buttons added on sites that enable the module for the first time from Twitter and Facebook to Facebook and X.

**Notes**

- It **does not** replace the existing Twitter button, for the reasons we discussed in #33118. This is less forceful on site owners.
- The new Share_X class extends the existing Twitter class so we can rely on that for now.
- At some point in the future, we may be able to merge the 2, once the Twitter brand is more deprecated than it is today.

### Other information:


## Jetpack product discussion

No discussion on this, but some previous related PRs:

- https://github.com/Automattic/social-logos/pull/126
- #33118

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

1. Go to Jetpack > Settings > Social and enable sharing buttons.
2. Go to Settings > Sharing and add a Twitter and an X sharing button.
3. They should both look good in wp-admin and on the frontend.
4. Clicking the button should work too.

> **Note**
> Matching Calypso PR: https://github.com/Automattic/wp-calypso/pull/81835
